radio/src/gui/colorlcd/widget_settings.cpp

```diff
@@ -101,6 +102,7 @@ WidgetSettings::WidgetSettings(Window* parent, Widget* widget) :
         new ModelTextEdit(line, rect_t{},
                           widget->getOptionValue(optIdx)->stringValue,
                           sizeof(widget->getOptionValue(optIdx)->stringValue));
+        SET_DIRTY();
```

Comment:
This does not work here. With your change, the widget will always be marked as dirty.
The ModelTextEdit takes care of marking the model data dirty when the text is changed by itself. You can just remove this here.